app/controllers/application_controller.rb

```diff
@@ -1,3 +1,10 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :set_page_number
+
+  private
+
+  def set_page_number
+    params[:page] = params[:page].present? ? params[:page].to_i : 1
+  end
```

Comment:
Nop, we shouldn't have any logic related to pagination in the controller, anything related to pagination should be passed to the service from the controller, that logic can lead to complexities in the future.